src/pages/MainPage/MainPage.js

```diff
@@ -1,9 +1,230 @@
+    const length = Number(list.map(item => item.insideList.length));
+
+    if (!insideCount < length && insideCount > 1) {
+      this.setState({
+        insideCount: insideCount - 1,
+      });
+
+      this.insideMargin += 1562;
+    }
+  };
+
+  componentDidMount() {
+    fetch('http://localhost:3007/data/mainData-songhyun.json')
+      .then(res => res.json())
+      .then(lists => {
+        const init = {
+          insideList: [],
+          eventList: [],
+          collectionList: [],
+        };
+
+        lists.forEach(item => {
+          const obj = {
+            inside: 'insideList',
+            collection: 'collectionList',
+            event: 'eventList',
+          };
+          init[obj[item.category]].push(item);
+        });
+        this.setState({ list: [init] });
+      });
+  }
   render() {
-    return <div></div>;
+    const { eventCount, insideCount, list } = this.state;
+    const saleLength = list.map(item => item.eventList.length);
+    const insideLength = list.map(item => item.insideList.length);
+
+    const saleStyle = {
+      marginLeft: this.saleMargin,
+    };
+    const insideStyle = {
+      marginLeft: this.insideMargin,
+    };
+
+    return (
+      <main>
+        <section className="newProductWrapper">
+          <div className="newProductContent">
+            <h2>Back to School</h2>
+            <strong>신상품 아이템 구매시,</strong>
+            <p>혜택1. 구매금액별 사은품 증정</p>
+            <p>혜택2. 신규가입시 1만원 할인</p>
+            <button type="button" className="plusBtn">
+              자세히 보기
+            </button>
+          </div>
+        </section>
+        <section className="newProductSaleWrapper">
+          <div>
+            <h3>당신만의 라페스타</h3>
+            <ol className="indications">
+              <li>
+                <button
+                  type="button"
+                  className="prevBtn"
+                  onClick={this.handlePrevClick}
+                  disabled={eventCount === 1}
+                >
+                  <i className="fas fa-chevron-left" />
+                </button>
+              </li>
+              <li className="startCount">
+                <span>{eventCount}</span>
+              </li>
+              <li className="center">
+                <span>/</span>
+              </li>
+              <li className="endCount">
+                <span>{Number(saleLength)}</span>
+              </li>
+              <li>
+                <button
+                  type="button"
+                  className="nextBtn"
+                  onClick={this.handleNextClick}
+                  disabled={eventCount === Number(saleLength)}
+                >
+                  <i className="fas fa-chevron-right" />
+                </button>
+              </li>
+            </ol>
+          </div>
+          <ul className="newProducts" style={saleStyle}>
+            {list.map(item =>
+              item.eventList.map(item => <MainItem key={item.id} item={item} />)
```

Comment:
그렇게 되면 이중 맵 돌릴 필요가 없습니다
```suggestion
            {eventList.map(item => <MainItem key={item.id} item={item} />)
```